src/github_repo_request_local.py

```diff
@@ -101,27 +220,74 @@ def list_test_files(directory, excluded_extensions):
                 capture_output=True,
             )
             logger.info(f"Successfully cloned the repo: {repo_name}")
+
+            # Fetch the last commit hash and store it in the DataFrame
+            last_commit_hash = get_last_commit_hash(clone_dir)
+            df.at[index, "last_commit_hash"] = last_commit_hash
+
+            # On successful clone, increment the processed_count
+            processed_count += 1
+
         except subprocess.CalledProcessError as e:
             logger.error(
                 f"Failed to clone the repo: {repo_name}." f" " f"Exception: {e}"
             )
             df.at[index, "testfilecountlocal"] = 0
+
+            # Even on failure, consider it processed for this batch
+            processed_count += 1
+
             continue
 
-    test_file_names = list_test_files(clone_dir, args.exclude)
-    count = len(test_file_names)
-    df.at[index, "testfilecountlocal"] = count
-    formatted_names = "\n".join(test_file_names)
-    logger.info(f"Test file names for {repo_name}:\n{formatted_names}")
+    # Always attempt to fetch the last commit hash if not already fetched
+    if pd.isna(row["last_commit_hash"]):
+        last_commit_hash = get_last_commit_hash(clone_dir)
+        df.at[index, "last_commit_hash"] = last_commit_hash
 
-    # Save after each update
+    # Count test files if not already counted
+    if row["testfilecountlocal"] == -1:
+        test_file_names = list_test_files(clone_dir, args.exclude)
+        count = len(test_file_names)
+        df.at[index, "testfilecountlocal"] = count
+        formatted_names = "\n".join(test_file_names)
+        logger.info(f"Test file names for {repo_name}:\n{formatted_names}")
+
+    processed_count += 1
+
+    # Processed count increment and batch check
+    if processed_count >= BATCH_SIZE:
+        # Save the DataFrame to CSV
+        df.to_csv(updated_csv_path, index=False)
+        logger.info(f"Batch of {BATCH_SIZE} repositories processed. Progress saved.")
```

Comment:
How about adding the name of the csv file to the log message, along the lines of: `f" ...Progress saved in {updated_csv_path}."`

Reply:
This is done.